fifo_queue.go

```diff
@@ -219,3 +219,60 @@ func (st *FIFO) IsLocked() bool {
 
 	return st.isLocked
 }
+
+// MoveFrontWithId moves the element at index position to the front of the queue
+func (st *FIFO) MoveFrontWithId(index int) error {
+
+	if st.isLocked {
+		return NewQueueError(QueueErrorCodeLockedQueue, "The queue is locked")
+	}
+	st.rwmutex.Lock()
+	defer st.rwmutex.Unlock()
+
+	length := len(st.slice)
+	if length == 0 {
+		return NewQueueError(QueueErrorCodeEmptyQueue, "Empty queue")
+	}
+
+	if index == 0 {
+		return NewQueueError(QueueErrorCodeIndexFirstPosition, "Element already in first position")
+	}
+
+	if index > length {
+		return NewQueueError(QueueErrorCodeIndexOutOfBounds, "Index is out of bounds")
+	}
+
+	for i := index; i >= 1; i-- {
+		st.slice[i], st.slice[i-1] = st.slice[i-1], st.slice[i]
+	}
```

Comment:
Please add a comment for this block.

Reply:
nit: This line is 124 characters long, which makes it hard to read on narrow editors. How about we split this into two lines?

Opening a discussion, I personally like to write my code documentation as if every comment is a step for a recipe. The first sentence of every comment block should start with a verb and should not contain implementation details, so that it can be read as a step. If any implementation details are relevant to the following block of code, it should be included after the first sentence. To know if you have a recipe, try removing the code of a function and only leaving the comments. The comments that are left should read like a series of actions you need to take to perform the function.

Using the comment for this block of code, we can rewrite it as:

```
// Move the element all the way to the front of the queue.
// The element is moved one position at time using bubble sort algorithm.
```

Notice the first sentence sounds like an action, and does not include any implementations details. The second line goes a bit into the implementation details. It doesn't apply in this case, but in general, documentation for implementation details should answer the question _why?_ and not _what?_. _What_ is being done can be inferred from the code, but _why_ things are done the way they're done helps whoever reads your code understand the reasoning behind the implementation.

Reply:
Thank you so much for this explanation, I'll keep trying to improve commenting things out!